torch/sparse/semi_structured/__init__.py

```diff
@@ -0,0 +1,79 @@
+import torch
+from .semi_structured_sparse_tensor import SemiStructuredSparseTensor
+import random
+
+__all__ = [
+    "to_semi_structured_sparse",
+    "is_semi_structured_sparse",
+    "gen_semi_structured_sparse_mask",
+    "SemiStructuredSparseTensor",
+]
+
+def gen_semi_structured_sparse_mask(r, c, dtype=torch.float16, device="cuda"):
+    """
+    This function returns a 1:2 sparse matrix of size (r, c).
+    Note that this means this matrix will also be 2:4 and 4:8 sparse as well.
+    """
+
+    choices = [[0, 1], [1, 0]]
+
+    mask_entries = [random.choice(choices) for i in range(r * c // 2)]
+    # print(mask_entries)
+
+    return (
+        torch.tensor(mask_entries, dtype=dtype, device=device)
+        .reshape(r, c)
+        .contiguous()
+    )
+
+
+def is_semi_structured_sparse(tensor: torch.Tensor, zeros_per_block=2):
+    """
+    return wether a tensor is semi_structured sparse
+    """
+
+    if not tensor.is_contiguous():
+        raise Exception("Tensor is not contiguous")
+
+    block_size = 2 * zeros_per_block
+    contiguous_flattened = tensor.view(-1)
+    # okay if not the same tensor since values will be the same
+    block_tensor = contiguous_flattened.reshape(-1, block_size)
+    assert ((block_tensor == 0).sum(dim=1) == zeros_per_block).all()
+
+
+def to_semi_structured_sparse(
```

Comment:
We could eventually create a native function of this name so that users can do `torch.Tensor.to_sparse_semi_structured` just like they can do `torch.Tensor.to_sparse_csr`.

Can we rename this to `to_sparse_semi_structured` just to fit the existing API pattern of `to_sparse_coo`, `to_sparse_bsr`, etc.